modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloader.java

```diff
@@ -177,18 +179,30 @@ void updateTaskState() {
     }
 
     //visible for testing
-    int indexChunks(String name, InputStream is, int chunk, String expectedMd5, long timestamp) throws IOException {
+    int indexChunks(String name, InputStream is, int startChunk, String expectedMd5, long timestamp) throws IOException {
+        int chunk = startChunk;
         MessageDigest md = MessageDigests.md5();
         for (byte[] buf = getChunk(is); buf.length != 0; buf = getChunk(is)) {
             md.update(buf);
-            client.prepareIndex(DATABASES_INDEX).setId(name + "_" + chunk + "_" + timestamp)
-                .setCreate(true)
-                .setSource(XContentType.SMILE, "name", name, "chunk", chunk, "data", buf)
-                .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-                .setWaitForActiveShards(ActiveShardCount.ALL)
-                .get();
+            IndexRequest indexRequest = new IndexRequest(DATABASES_INDEX)
+                .id(name + "_" + chunk + "_" + timestamp)
+                .create(true)
+                .source(XContentType.SMILE, "name", name, "chunk", chunk, "data", buf)
+                .waitForActiveShards(ActiveShardCount.ALL);
```

Comment:
I wonder if we need to wait for active shards, I think I've added it as an attempt to solve some problems I've encountered. It can be removed probably?

Reply:
I think it is safe to remove this; if the primary shard isn't available then by default the index request waits for a minute for the primary shard get started. If the replica shard is missing then, index request will be executed and replica shard will catch up with the primary shard before be getting online.